pkg/data/management/role_data.go

```diff
@@ -637,6 +669,25 @@ func BootstrapAdmin(management *wrangler.Context) (string, error) {
 	return adminName, nil
 }
 
+func addPrincipalLabelToAdmin(admin *v3.User, management *wrangler.Context) (*v3.User, error) {
+	// admin may not have a principal yet. To get around a complex wait loop, gen the principal name as the provider does
+	principalName := localauth.GetLocalPrincipalID(admin)
+	encodedPrincipalID := base32.HexEncoding.WithPadding(base32.NoPadding).EncodeToString([]byte(principalName))
+	if len(encodedPrincipalID) > 63 {
+		encodedPrincipalID = encodedPrincipalID[:63]
+	}
```

Comment:
Since that logging is not necessary, I'd move this inside the main if statement, so that you don't generate the encoded principal ID for nothing, if the label already exists.

Reply:
Will remove the logging in next iteration.

Reply:
Fixed.